source/style-guide/markup/directives/sharedinclude.txt

```diff
@@ -0,0 +1,234 @@
+   `10gen/docs-shared <https://github.com/10gen/docs-shared/>`__
+   private GitHub repository.
+
+
+Setup
+-----
+
+You must add the URL from which to source the shared content in the
+``sharedinclude_root`` setting of the documentation repository's ``snooty.toml``
+configuration file. This should reference the unprocessed versions of the files,
+currently served from the ``raw.githubusercontent.com`` domain.
+
+If the shared content that your docs repository file references is on the
+``main`` branch of ``docs-snooty``, your ``snooty.toml`` configuration entry
+should resemble the following line:
+
+.. code-block::
+
+   sharedinclude_root = "https://raw.githubusercontent.com/10gen/docs-shared/main/"
+
+Snooty reports an error if you attempt to use ``sharedinclude`` without
+configuring this setting.
+
+
+Syntax
+------
+
+To include content from an external file, add the ``sharedinclude`` directive
+to the sourcing file. Specify the path of external file relative to the
+``docs-shared`` base directory, omitting the leading slash.
+
+For example, if the file path from the root of the ``docs-shared`` repository
+is ``drivers/compatibility-tables/c.rst``, the rST in the sourcing file
+should include the following code:
+
+.. code-block:: rst
+   :caption: sourcing_file.txt in a documentation repository
+   :copyable: false
+
+   .. _sharedinclude:: drivers/compatibility-tables/c.rst
+
+The external file can include replacement variables in the form of
+``|variable-name|``. This enables the sourcing file to customize segments
+of the page.
+
+For example, suppose the ``docs-shared`` file ``language.rst`` contained the
+following content with a replacement variable called ``|driver-name|``:
+
+.. code-block:: rst
+   :caption: language.rst in the docs-shared repository
+   :copyable: false
+
+   The following table shows the compatibility between different versions of
+   the |driver-name| driver and the |driver-language|:
+
+The sourcing file could resemble the following code:
+
+.. code-block:: rst
+   :caption: compatibility-table.txt sourcing file in the documentation repository
+   :copyable: false
+
+  .. _sharedinclude:: drivers/compatibility-tables/language.rst
+
+     .. replacement:: driver-name
+
+        PyMongo
+
+     .. replacement:: driver-language
+
+        Python language
+
+The published page displays the following text:
+
+.. code-block:: rst
+   :copyable: false
+
+   The following table shows the compatibility between different versions of
+   the PyMongo driver and the Python language:
+
+Snooty reports a warning or error if the sourcing file is missing any
+replacement variables present in the external file.
+
+Placeholders in Inline Context
+``````````````````````````````
+
+An **inline context** describes the positioning of placeholders as adjacent to
+other paragraph text. You can only replace variable placeholders in an inline
+context with values that contain markup elements which includes any of the
+following types:
```

Comment:
[Wordsmith suggestion]

Could we rewrite this sentence as something like the following to try and simplify the language:

> You can replace variable placeholders in an inline context with values that contain only the following markup elements:

Reply:
Thanks for the suggestion. I think this may lose the nuance that the list is not complete, but rather includes the types in the list. Will try a rewrite.